src/api/Base.js

```diff
@@ -86,6 +90,17 @@ class Base {
         this.destroyed = true;
     }
 
+    /**
+     * Update upload host with reachable url
+     *
+     * @return {Promise<*>}
+     */
+    async updateReachableUploadHost(): Promise<*> {
+        this.uploadHost = await this.uploadsReachability.getReachableUploadHost();
+        this.options.uploadHost = this.uploadHost;
```

Comment:
Any idea why we need to set `apiHost`, `cache` and `uploadHost` in `this.options` when we have all the data flattened in Base @priyajeet?

A little confusing to have to update it in two places.

Reply:
Actually, I think we don't need `this.options`. And `Xhr` doesn't accept apiHost, cache or uploadHost in the constructor `options` param. Fixing that

Reply:
nvm, I'm not sure, keeping it as is for now and adding a comment